### PR TITLE
Use working pre-built sccache binary

### PR DESCRIPTION
**Pull request by huydhn**
From https://github.com/pytorch/pytorch/pull/95938 where a new Docker image build fails to start sccache. This issue starts to happen today (Mar 3rd).   The server fails to start with a cryptic `sccache: error: Invalid argument (os error 22)`

```
=================== sccache compilation log ===================
ERROR 2023-03-03T20:31:14Z: sccache::server: failed to start server: Invalid argument (os error 22)

sccache: error: Invalid argument (os error 22)

=========== If your build fails, please take a look at the log above for possible reasons ===========

+ sccache --show-stats
sccache: error: Connection to server timed out
```

I don't have a good explanation for this yet.  The version of sccache we build from https://github.com/pytorch/sccache is ancient.  If I start to build the exact same version on Ubuntu Docker image now, the issue will manifest.  But the older binary built only few days ago https://hud.pytorch.org/pytorch/pytorch/commit/e50ff3fcdb3890ce3bbab99e60b1c27ff49be2af works without any issue.  So I fix sccache binary to that version instead of rebuilding it every time in the image as a temporary mitigation while trying to root cause this further.

### CODE REVIEW

**Comment by huydhn** [comment]
Some above steps fails because I forgot to grant read access to the binary from S3.  Once pull workflow finishes, retrying will work

**Comment by huydhn** [comment]
I probably need someone with Rust knowledge to debug the issue further.  So here is my finding so far.  The server failed to start because it couldn't bind to the port (invalid argument, errno 22).  Our sccache fork uses this old crate https://crates.io/crates/tokio-tcp/0.1.4 for the job:

```
use tokio_tcp::TcpListener;
...
let addr = SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), port);
let listener = TcpListener::bind(&SocketAddr::V4(addr))?; <--- FAILING HERE
...
```

The code has not changed, so I still couldn't tell why it starts failing now. Confirm that there is no process running hoarding sccache port.  So the only thing left from https://man7.org/linux/man-pages/man2/bind.2.html is:

```
EINVAL addrlen is wrong, or addr is not a valid address for this socket's domain.
```

But AFIAK, the code looks reasonable, binding to "127.0.0.1:4226".  I already create a task [T136185557](https://www.internalfb.com/intern/tasks/?t=136185557) to update sccache version, but lack the Rust knowledge to finish it.

**Comment by huydhn** [comment]
@ZainRizvi Let's get this merge to unblock your PR and others like https://github.com/pytorch/pytorch/pull/95896 to update triton pinned commit.  Anything touching Docker image would be blocked otherwise.

@ngimel Those failed builds due to sccache in https://github.com/pytorch/pytorch/pull/95896 would be fixed once this PR is merged.

**Comment by ngimel** [comment]
@huydhn great, thanks! #95896 also needs CMake v 3.20 (we currently have v 3.17), can we update that?

**Comment by huydhn** [comment]
> @huydhn great, thanks! #95896 also needs CMake v 3.20 (we currently have v 3.17), can we update that?

FYI, the PR to upgrade cmake version is here https://github.com/pytorch/builder/pull/1331

**Review by ZainRizvi — APPROVED** [review]
Thanks for debugging this!

Any idea who owns our sccache fork? It would be good to raise this issue to them (if it even has an owner)

**Comment by huydhn** [comment]
> Any idea who owns our sccache fork? It would be good to raise this issue to them (if it even has an owner)

Ed was last seen updating it https://github.com/pytorch/sccache/commits/master, but I think this is likely that we are the owner.  Let's me bring this up for discussion this week.

**Comment by malfet** [comment]
We should have a followup task for this one, to resort back to the source builds

**Pull request closed without merging**
